tests/pytests/unit/utils/jinja/test_custom_extensions.py

```diff
@@ -0,0 +1,1088 @@
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "None"
+
+    rendered = render_jinja_tmpl(
+        "{{ '127.0.0.1' | ipv4(options='lo') }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "127.0.0.1"
+
+
+def test_ipv6(minion_opts, local_salt):
+    """
+    Test the `ipv6` Jinja filter.
+    """
+    rendered = render_jinja_tmpl(
+        "{{ '192.168.0.1' | ipv6 }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "None"
+
+    rendered = render_jinja_tmpl(
+        "{{ 'random' | ipv6 }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "None"
+
+    # returns the standard format value
+    rendered = render_jinja_tmpl(
+        "{{ 'FE80:0:0::0' | ipv6 }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "fe80::"
+
+    # fe80:: is link local therefore will be returned
+    rendered = render_jinja_tmpl(
+        "{{ 'fe80::' | ipv6(options='ll') }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "fe80::"
+
+    # fe80:: is not loopback
+    rendered = render_jinja_tmpl(
+        "{{ 'fe80::' | ipv6(options='lo') }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "None"
+
+    # returns only IPv6 addresses in the list
+    rendered = render_jinja_tmpl(
+        "{{ ['fe80::', '192.168.0.1'] | ipv6 | join(', ') }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "fe80::"
+
+    rendered = render_jinja_tmpl(
+        "{{ ['fe80::', '::'] | ipv6 | join(', ') }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "fe80::, ::"
+
+
+def test_network_hosts(minion_opts, local_salt):
+    """
+    Test the `network_hosts` Jinja filter.
+    """
+    rendered = render_jinja_tmpl(
+        "{{ '192.168.0.1/30' | network_hosts | join(', ') }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "192.168.0.1, 192.168.0.2"
+
+
+def test_network_size(minion_opts, local_salt):
+    """
+    Test the `network_size` Jinja filter.
+    """
+    rendered = render_jinja_tmpl(
+        "{{ '192.168.0.1' | network_size }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "1"
+
+    rendered = render_jinja_tmpl(
+        "{{ '192.168.0.1/8' | network_size }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "16777216"
+
+
+@pytest.mark.requires_network
+def test_http_query(minion_opts, local_salt):
+    """
+    Test the `http_query` Jinja filter.
+    """
+    urls = (
+        # These cannot be HTTPS urls since urllib2 chokes on those
```

Comment:
I know these were copied over, migrated, but are we still using urllib2 in our code?